code/datums/components/cleaner.dm

```diff
@@ -0,0 +1,88 @@
+/**
+ * Component that can be used to clean things.
+ * Takes care of duration, cleaning skill and special cleaning interactions.
+ * A callback can be set by the datum holding the cleaner to add custom functionality.
+ * Soap uses a callback to decrease the amount of uses it has left after cleaning for example.
+ */
+/datum/component/cleaner
+	/// The time it takes to clean something, without reductions from the cleaning skill modifier.
+	var/base_cleaning_duration
+	/// Offsets the cleaning duration modifier that you get from your cleaning skill, the duration won't be modified to be more than the base duration.
+	var/skill_duration_modifier_offset
+	/// Determines what this cleaner can wash off, [the available options are found here](code/__DEFINES/cleaning.html).
+	var/cleaning_strength
+	/// Multiplies the cleaning skill experience gained from cleaning.
+	var/experience_gain_modifier
+	/// Gets called when something is successfully cleaned.
+	var/datum/callback/on_cleaned_callback
+	/// Gets invoked asynchronously by the signal handler
+	var/datum/callback/cleaning_proc
```

Comment:
Do we need this var or can we just ``INVOKE_ASYNC(src, .proc/clean, source, target, user)`` when needed?